tests/migrations/test_autodetector.py

```diff
@@ -2720,6 +2720,112 @@ def test_add_constraints(self):
             changes, "testapp", 0, 0, model_name="author", constraint=added_constraint
         )
 
+    def test_add_constraints_with_new_model(self):
+        book = ModelState(
+            "otherapp",
+            "Book",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("title", models.CharField(max_length=200)),
+            ],
+        )
+
+        category = ModelState(
+            "otherapp",
+            "Category",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("name", models.CharField(max_length=200)),
+            ],
+        )
+
+        book_with_unique_title_and_category = ModelState(
+            "otherapp",
+            "Book",
+            [
+                ("id", models.AutoField(primary_key=True)),
+                ("title", models.CharField(max_length=200)),
+                ("category", models.ForeignKey("otherapp.Category", models.CASCADE)),
+            ],
+            {
+                "constraints": [
+                    models.UniqueConstraint(
+                        fields=["title", "category"],
+                        name="unique_title_category",
+                    )
+                ]
+            },
+        )
```

Comment:
Wouldn't we be able to re-use existing `ModelState` class attributes?
`django.migrations.test_autodetector.AutodetectorTests` has so many! 😄 

Perhaps with:
```
previous state: [self.book_with_no_author]
to state: [self.book_indexes, self.author_empty]
```
🤔 
(I haven't tested it, it's just from a quick look at the class.)

(and same remark for `test_add_index_with_new_model`)